qwen_agent/llm/oai.py

```
@@ -0,0 +1,105 @@
+            if delta_stream:
+                for chunk in response:
+                    if chunk.choices and hasattr(chunk.choices[0].delta, 'content') and chunk.choices[0].delta.content:
+                        yield [Message(ASSISTANT, chunk.choices[0].delta.content)]
+            else:
+                full_response = ''
+                for chunk in response:
+                    if chunk.choices and hasattr(chunk.choices[0].delta, 'content') and chunk.choices[0].delta.content:
+                        full_response += chunk.choices[0].delta.content
+                        yield [Message(ASSISTANT, full_response)]
+        except OpenAIError as ex:
+            raise ModelServiceError(exception=ex)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)